app/routes/logout.tsx

```diff
@@ -0,0 +1,17 @@
+import { redirect } from "@remix-run/node";
+import type { ActionFunction } from "@remix-run/node";
+import { destroySession, getSession } from "~/session.server";
+
+export const action: ActionFunction = async ({ request }) => {
+  let session = await getSession(request.headers.get("Cookie"));
+  session.unset("idToken");
+
+  return redirect("/", {
+    headers: { "Set-Cookie": await destroySession(session) },
+  });
+};
+
+export const loader = () => {
+  // Redirect to the home page
+  return redirect("/");
```

Comment:
You'll notice that this route doesn't actually render a component! If someone navigates to it directly, they'll get redirected to the home page.

That concludes the tour of the authentication flow, but I want to point out a few more things.